### PR TITLE
feat(api): Add `AuthModule`

### LINKED ISSUE #11: API Auth Module
Requirements:
- [ ] [`express-session`](https://www.npmjs.com/package/express-session)
- [ ] [`connect-redis`](https://www.npmjs.com/package/connect-redis)
- [ ] [`ioredis`](https://www.npmjs.com/package/ioredis)
- [ ] [`passport`](https://www.npmjs.com/package/passport), [`@nestjs/passport`](https://www.npmjs.com/package/@nestjs/passport)
- [ ] [`bcryptjs`](https://www.npmjs.com/package/bcryptjs)
- [ ] [`passport-local`](https://www.npmjs.com/package/passport-local)

Additional features:
- [ ] Google Authentication
- [ ] Facebook Authentication
- [ ] GitHub Authentication

### PR DESCRIPTION
**Pull request by maciejopalinski**

### CODE REVIEW

File: apps/api/src/auth/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+import { User } from '@ecoside/database';
+import {
+    ActivateAccountDto,
+    LoginDto,
+    RegisterDto,
+    RequestPasswordResetDto,
+    ResetPasswordDto,
+} from '@ecoside/dto';
+import { AuthService, CookieGuard, LoginGuard, Request, ReqUser } from '.';
```
Comment (JakubZojdzik) [inline apps/api/src/auth/auth.controller.ts:28]: Isn't it better to make it multiline?

File: apps/api/src/auth/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    async register(dto: RegisterDto) {
+        const user = this.users.create(dto);
+        user.password_hash = hashSync(dto.password);
+
+        try {
+            await this.users.save(user);
+        } catch {
+            throw new ConflictException(
```
Comment (JakubZojdzik) [inline apps/api/src/auth/auth.service.ts:53]: Is it only error that can occur here?

**Review by JakubZojdzik — DISMISSED** [review]
Look's good, but you have to explain me some parts

File: apps/api/src/auth/auth.service.ts (hunk 2 above)
Comment (maciejopalinski) [reply apps/api/src/auth/auth.service.ts:53]: No there are many more errors possible but checking for error codes everytime you save something is a hassle to do. I thought about it and maybe there is a way to make our own Repository class that would handle all the other errors.

File: apps/api/src/auth/auth.controller.ts (hunk 1 above)
Comment (maciejopalinski) [reply apps/api/src/auth/auth.controller.ts:28]: Prettier decided for me :P
That's why we have Prettier

File: apps/api/src/auth/auth.service.ts (hunk 2 above)
Comment (JakubZojdzik) [reply apps/api/src/auth/auth.service.ts:53]: So I'm not sure if returning 'Account with that email address already exists' error in 54 line is good.

Comment (maciejopalinski) [reply apps/api/src/auth/auth.service.ts:53]: Yeeeeah, but 99% of the times the error is regarding the unique constraint on email column. Other cases that .save() throws an error is when there is a issue with schema. In production it is basically the only error that would occur here.
It is good enough but I will try to implement some kind of custom repository / error handler.

**Review by JakubZojdzik — APPROVED** [review]

**Pull request merged**